src/main/java/com/microsoft/graph/authentication/TokenCredentialAuthProvider.java

```diff
@@ -0,0 +1,134 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.exceptions.Error;
+import com.microsoft.graph.exceptions.ErrorConstants;
+import com.microsoft.graph.httpcore.IHttpRequest;
+import okhttp3.Request;
+import com.microsoft.graph.exceptions.ErrorConstants.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+
+import javax.annotation.Nonnull;
+
+public class TokenCredentialAuthProvider implements ICoreAuthenticationProvider , IAuthenticationProvider {
+
+    //TokenCredential expected from user
+    private TokenCredential tokenCredential;
+    //Context options which can be optionally set by the user
+    private TokenRequestContext context;
+    // maximum delay to wait for token obtention
+    private Duration tokenBlockTimeout;
+
+    
+    //TODO: Upon further review from Peter, this should include a Null check in the case that this is the first request
+    //There should be just one constructor as scopes should probably not be optional
+    /**
+     * Creates an Authentication provider using a passed in TokenCredential
+     *
+     * @param tokenCredential Credential object inheriting the TokenCredential interface used to instantiate the Auth Provider
+     * @throws AuthenticationException exception occurs if the TokenCredential parameter is null
+     */
+    public  TokenCredentialAuthProvider(@Nonnull final TokenCredential tokenCredential) throws AuthenticationException {
+        if(tokenCredential == null) {
+            throw new AuthenticationException(new Error(Codes.InvalidArgument,
+                    String.format(Messages.NullParameter, "TokenCredential"))
+                    ,new IllegalArgumentException());
+        }
+
+        this.tokenCredential = tokenCredential;
+        this.context = new TokenRequestContext();
+        this.tokenBlockTimeout = Duration.ofMinutes(10);
+    }
+
+    /**
+     * Creates an Authentication provider using a TokenCredential and list of scopes
+     *
+     * @param tokenCredential Credential object inheriting the TokenCredential interface used to instantiate the Auth Provider
+     * @param scopes Specified desired scopes of the Auth Provider
+     * @throws AuthenticationException exception occurs if the TokenCredential parameter is null
+     */
+    public TokenCredentialAuthProvider(@Nonnull final List<String> scopes, @Nonnull final TokenCredential tokenCredential) throws AuthenticationException {
```

Comment:
What happens when using a scope like 'https://graph.microsoft.com/.default ' in case of client credentials flow?

Reply:
The scopes returned are defined by the application scopes consented via the Azure Portal.

Reply:
See https://github.com/microsoftgraph/msgraph-sdk-java-core/pull/42#discussion_r561181698 but I added it as a default value in case the scopes are not being passed by the consumers.